### ISSUE TITLE
Janky scrolling after bringing app from background on iPhone X

### ISSUE THREAD

**Issue by knopp** [flutter/flutter#24282]
This can be reproduced with flutter gallery, but seems to be happening with pretty much any flutter application. 

When bringing app from background, the scrolling is extremely janky (below 30 fps, even on iPhone X) for maybe 10 seconds, after which it gets resolved and is smooth again.

```
[✓] Flutter (Channel master, v0.11.4-pre.30, on Mac OS X 10.14 18A384a, locale en-US)
    • Flutter version 0.11.4-pre.30 at /Users/Matej/Projects/flutter/flutter
    • Framework revision aa65fa2fa1 (5 hours ago), 2018-11-12 23:47:16 -0800
    • Engine revision ea4bbbd440
    • Dart version 2.1.0 (build 2.1.0-dev.9.4 9c07fb64c4)

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).
      If Android SDK has been installed to a custom location, set $ANDROID_HOME to that location.
      You may also want to add it to your PATH environment variable.


[✓] iOS toolchain - develop for iOS devices (Xcode 10.0)
    • Xcode at /Applications/Xcode-beta.app/Contents/Developer
    • Xcode 10.0, Build version 10L221o
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.3

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).

[!] IntelliJ IDEA Community Edition (version 2018.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.io/intellij-setup/#installing-the-plugins

[✓] VS Code (version 1.29.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 2.20.0

[✓] Connected device (2 available)
    • Matej’s iPhone X • f7e7ead1f9b5f3e2574cfbd9bc264fbcd5772a6c • ios • iOS 11.4.1
    • iPhone 5s        • BB2ADE05-BD55-4573-AB1D-0E932E53B592     • ios • iOS 12.0 (simulator)
```

**Comment by Panl** [comment]
the same problem on my iPhone X

**Comment by knopp** [comment]
This seems to be massively improved in current master, I'm guessing due shader cache finally being enabled?

**Comment by knopp** [comment]
This seems to have regressed significantly in current master.

**Comment by darshankawar** [comment]
Hi @knopp,
I tried a sample listview app on iPhone SE using latest stable (1.20.2) and didn't see the jank in scrolling after switching between apps and opening the app from background, unless I missed any step to replicate.

[24282.mov.zip](https://github.com/flutter/flutter/files/5154964/24282.mov.zip)

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 1.20.2, on Mac OS X 10.15.2 19C57, locale en-US)
    • Flutter version 1.20.2 at
      /Users/deeptibelsare/Documents/Fluttersdk/flutter
    • Framework revision bbfbf1770c (3 weeks ago), 2020-08-13 08:33:09 -0700
    • Engine revision 9d5b21729f
    • Dart version 2.9.1

[!] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/deeptibelsare/Library/Android/sdk
    • Platform android-30, build-tools 29.0.3
    • ANDROID_HOME = /Users/deeptibelsare/Library/Android/sdk
    • ANDROID_SDK_ROOT = /Users/deeptibelsare/Library/Android/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)
    ✗ Android license status unknown.
      Try re-installing or updating your Android SDK Manager.
      See https://developer.android.com/studio/#downloads or visit
      https://flutter.dev/docs/get-started/install/macos#android-setup for
      detailed instructions.

[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.6, Build version 11E708
    • CocoaPods version 1.9.3

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)

[✓] Connected device (1 available)
    • SM A260G (mobile) • 5200763ebcfa861f • android-arm • Android 8.1.0 (API
      27)

! Doctor found issues in 1 category.
```

</details>

Can you confirm if this is still an issue at your end ?
Thanks.

**Comment by knopp** [comment]
I *think* this was fixed in #31086.

**Comment by darshankawar** [comment]
Thanks @knopp. I am going ahead and closing this issue as fixed. If you disagree, write in comments and I'll reopen.
Thanks.
